README.md

```diff
@@ -1,94 +1,87 @@
 
+# Note that the ‘requirements.txt’ file already lists a version of ‘ops’. 
 
-## Getting started
+# Note that ‘src/charm.py’ already uses ops constructs
+# to create a charm class ‘MyNewCharmCharm’, observe Juju events, and pair them to event handlers:
+#
+# class MyNewCharmCharm(ops.CharmBase):
+#     """Charm the service."""
+# 
+#     def __init__(self, *args):
+#         super().__init__(*args)
+#         self.framework.observe(self.on['httpbin'].pebble_ready, self._on_httpbin_pebble_ready)
+#         self.framework.observe(self.on.config_changed, self._on_config_changed)
+# ...
 
-A package of operator code is called a charmed operator or simply "charm".
-You'll use [charmcraft](https://juju.is/docs/sdk/install-charmcraft) to
-register your charm name and publish it when you are ready. You can follow one
-of our [charming tutorials](https://juju.is/docs/sdk/tutorials) to get started
-writing your first charm.
+# Edit the files to make the charm your own.
 
+# When you’re done, pack the example charm into a .charm file:
+charmcraft pack
 
-## Testing your charms
+# If you haven’t edited the name of the charm,
+# this will yield a file called ‘my-new-charm_ubuntu-22.04-amd64.charm’
+# (the architecture bit may be different depending on your system's architecture).
 
-The framework provides a testing harness, so you can ensure that your charm
-does the right thing in different scenarios, without having to create
-a full deployment. Our [API documentation](https://ops.readthedocs.io/en/latest/#module-ops.testing)
-has the details, including this example:
+# Use the name from the packing step and the resource from the ‘metadata.yaml’
+# to deploy your example charm to your local MicroK8s cloud, for example:
+juju deploy ./my-new-charm_ubuntu-22.04-amd64.charm --resource httpbin-image=kennethreitz/httpbin
 
-```python
-class TestCharm(unittest.TestCase):
-    def test_foo(self):
-        harness = Harness(MyCharm)
-        self.addCleanup(harness.cleanup)  # always clean up after ourselves
+# Congratulations, you’ve just built your first charm!
 
-        # Instantiate the charm and trigger events that Juju would on startup
-        harness.begin_with_initial_hooks()
+# Now you can go back and edit the files to make the charm your own.
 
-        # Update charm config and trigger config-changed
-        harness.update_config({'log_level': 'warn'})
+# Once you’re done, come back to Charmcraft to publish and release your charm on Charmhub!
 
-        # Check that charm properly handled config-changed, for example,
-        # the charm added the correct Pebble layer
-        plan = harness.get_container_pebble_plan('prometheus')
-        self.assertIn('--log.level=warn', plan.services['prometheus'].command)
+# Clean up:
+# - delete the vm:
+multipass delete --purge my-vm
+# - uninstall multipass:
+snap remove multipass
 ```
+For more, see our user docs: https://juju.is/docs/sdk 
 
+## Contribute to Ops
```

Comment:
I'm not sure that "talk to us" is necessarily the best heading, but it's better than "contribute to ops". We're not providing these links just for people that might contribute - we're providing them to users of ops (a far bigger group).

Reply:
I thought about this for a long time. 

The reason I decided for a simple split into Use and Contribute -- with Contribute going progressively from talking to us to filing a bug to making a change, in the code or the docs -- is because I am not satisfied with the usual split into "End users" and "Developers": 

- Developers are also Users; 
- Users can also be developers as they can in principle also contribute to both the code and the docs (in fact, that's the whole premise behind open source, right?); and 
- imo contribution is not just submitting a PR to the code or the docs but also filing a bug and in fact even just talking to us, because that provides implicit feedback too and it's often the first stepping stone towards filing an issue and submitting a change.

However, I recognize that the decision is a bit unorthodox, so happy to replace the single Contribute section with three sections:
- Talk to us
- File an issue
- Make a change

The decision is ultimately up to you, the Ops project owners, so I will make whatever change you request. However, (1) it would ultimately be good to have uniformity across the various Juju world projects and (2) the pattern I am am currently recommending is the Use - Contribute one I mentioned, for the reasons mentioned above.